builtin/nomad/task.go

```diff
@@ -286,9 +287,106 @@ func (p *TaskLauncher) StartTask(
 func (p *TaskLauncher) WatchTask(
 	ctx context.Context,
 	log hclog.Logger,
+	ui terminal.UI,
 	ti *TaskInfo,
 ) (*component.TaskResult, error) {
-	return nil, status.Errorf(codes.Unimplemented, "WatchTask not implemented")
+	// We'll query for the allocation in the namespace of our task launcher
+	queryOpts := &api.QueryOptions{Namespace: p.config.Namespace}
+
+	// Accumulate our result on this
+	var result component.TaskResult
+	client, err := getNomadClient()
+	if err != nil {
+		return nil, err
+	}
+	allocs, _, err := client.Jobs().Allocations(ti.Id, true, queryOpts)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(allocs) != 1 {
+		log.Error("Invalid # of allocs for ODR job.")
+		return nil, status.Error(codes.Internal, "there should be one allocation in the job")
```

Comment:
I'd include the length here too for extra debug information in both the log an error:

> log.Error("Invalid ...", "total_allocs", len(allocs))

For example